python/cuml/cuml/internals/base_helpers.py

```diff
@@ -54,84 +53,40 @@ def _wrap_attribute(class_name: str, attribute_name: str, attribute, **kwargs):
-        and (
-            p.kind == Parameter.POSITIONAL_ONLY
-            or p.kind == Parameter.POSITIONAL_OR_KEYWORD
-        )
-    ]
-
-    assert len(incorrect_params) == 0, (
-        "Error in `{}`!. Positional arguments for estimators (that derive "
-        "from `Base`) have been deprecated but parameters '{}' can still "
-        "be used as positional arguments. Please specify all parameters "
-        "after `self` as keyword only by using the `*` argument"
-    ).format(attribute.__qualname__, ", ".join(incorrect_params))
-
-    return _deprecate_pos_args(**kwargs)(attribute)
-
-
 class BaseMetaClass(type):
     """
-    Metaclass for all estimators in cuML. This metaclass will get called for
-    estimators deriving from `cuml.common.Base` as well as
-    `cuml.dask.common.BaseEstimator`. It serves 2 primary functions:
-
-     1. Set the `@_deprecate_pos_args()` decorator on all `__init__` functions
-     2. Wrap any functions and properties in the API decorators
-        [`cuml.common.Base` only]
+    Metaclass for all estimators in cuML.
 
+    This metaclass will get called for estimators deriving from `cuml.common.Base`
+    as well as `cuml.dask.common.BaseEstimator`. It automatically wraps methods and
+    properties in the API decorators (`cuml.common.Base` only).
     """
 
-    def __new__(cls, classname, bases, classDict):
-
-        is_dask_module = classDict["__module__"].startswith("cuml.dask")
-
-        for attributeName, attribute in classDict.items():
-
-            # If attributeName is `__init__`, wrap in the decorator to
-            # deprecate positional args
-            if attributeName == "__init__":
-                attribute = _check_and_wrap_init(attribute, version="21.06")
-                classDict[attributeName] = attribute
-
-            # For now, skip all additional processing if we are a dask
-            # estimator
-            if is_dask_module:
-                continue
-
-            # Must be a function
-            if callable(attribute):
-
-                classDict[attributeName] = _wrap_attribute(
-                    classname, attributeName, attribute
-                )
+    def __new__(cls, classname, bases, namespace):
+        # Skip wrapping methods in dask estimators
+        if not namespace["__module__"].startswith("cuml.dask"):
+            for name, attribute in namespace.items():
+                if callable(attribute):
+                    # Wrap method
+                    namespace[name] = _wrap_attribute(
+                        classname, name, attribute
+                    )
 
-            elif isinstance(attribute, property):
-                # Need to wrap the getter if it exists
-                if hasattr(attribute, "fget") and attribute.fget is not None:
-                    classDict[attributeName] = attribute.getter(
+                elif (
+                    isinstance(attribute, property)
+                    and attribute.fget is not None
```

Comment:
I assume all `property`s have a `fget` attribute and the old `hasattr` check was superfluous?